cellprofiler/modules/watershed.py

```diff
@@ -116,8 +125,15 @@ def run(self, workspace):
         if self.operation.value == "Distance":
             original_shape = x_data.shape
 
-            if x.volumetric:
-                x_data = skimage.transform.resize(x_data, (original_shape[0], 256, 256), order=0, mode="edge")
+            factor = self.downsample.value
+
+            if factor > 1:
+                if x.volumetric:
+                    factors = (factor, factor, factor)
+                else:
+                    factors = (factor, factor)
+
+                x_data = skimage.transform.downscale_local_mean(x_data, factors)
```

Comment:
This operation does not produce a binary image when provided a binary image as input.